connectors/source.py

```diff
@@ -168,14 +208,26 @@ def dependencies_satisfied(self, field):
             if dependency["field"] not in self._config:
                 # cannot check dependency if field does not exist
                 raise ConfigurableFieldDependencyError(
-                    f'Configuration `{field.label}` depends on configuration `{dependency["field"]}`, but it does not exist.'
+                    f'`{field.label}` depends on configuration `{dependency["field"]}`, but it does not exist.'
                 )
 
             if self._config[dependency["field"]].value != dependency["value"]:
                 return False
 
         return True
 
+    def is_value_empty(self, field):
```

Comment:
This method can be moved to `Field` class. `dependencies_satisfied` as well.

Reply:
> This method can be moved to Field class

👍🏻 
The same is true for `validate_field`. I'll move both over.

> dependencies_satisfied as well.

`dependencies_satisfied` needs access to other configuration fields to make comparisons, so I think it needs to be here. Unless there is a way to access them from within Field?

Reply:
I've added this to my to-do list on the epic: https://github.com/elastic/enterprise-search-team/issues/3065 (under Connectors)